src/bin/upload-perf.rs

```diff
@@ -18,33 +50,89 @@ fn main() {
         Ok(file) => file,
     };
 
+    let upload_flag = &args[2];
```

Comment:
nit: How about flipping the order of the branch and upload_flag arguments, then make upload_flag optional (where missing == "no-upload")